torchmetrics/text/bleu.py

```diff
@@ -91,17 +91,22 @@ def __init__(
         self.add_state("denominator", torch.zeros(self.n_gram), dist_reduce_fx="sum")
 
     def update(  # type: ignore
-        self, reference_corpus: Sequence[Sequence[Sequence[str]]], translate_corpus: Sequence[Sequence[str]]
+        self, reference_corpus: Sequence[Sequence[str]], translate_corpus: Sequence[str]
     ) -> None:
         """Compute Precision Scores.
 
         Args:
             reference_corpus: An iterable of iterables of reference corpus
             translate_corpus: An iterable of machine translated corpus
         """
+        reference_corpus_: Sequence[Sequence[Sequence[str]]] = [
+            [line.split() for line in reference] for reference in reference_corpus
+        ]
+        translate_corpus_: Sequence[Sequence[str]] = [line.split() for line in translate_corpus]
```

Comment:
lets move this code to the `_blue_score_update` function. Then we do not need to have duplicates.